### PR TITLE
MF-571 - Fix user login and refactor User

### PR DESCRIPTION
**Pull request by darkodraskovic**
Further modularize code and reuse modules. Fix bugs, like logout button not working.

### CODE REVIEW

File: ui/src/HttpMF.elm [hunk 1]
```diff
@@ -8,6 +8,15 @@ import Json.Encode as E
 import Url.Builder as B
 
 
+url =
+    { base = "http://localhost"
```
Comment (drasko) [inline ui/src/HttpMF.elm:12]: Take care that this will have to be configurable through ENV variable to work in Docker, I think.

**Review by drasko — APPROVED** [review]
LGTM

File: ui/src/HttpMF.elm (hunk 1 above)
Comment (darkodraskovic) [reply ui/src/HttpMF.elm:12]: Ok. Sure. We'll add a variable. We'll probably need some bash script to run the app and to set env vars or something similar.

**Review by blokovi — APPROVED** [review]
LGTM

**Pull request merged**